katz/src/main/kotlin/katz/NonEmptyList.kt

```diff
@@ -66,6 +66,6 @@ class NonEmptyList<out A> private constructor(
     }
 
     companion object Factory {
-        fun <A> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
+        fun <A: Any> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
     }
```

Comment:
The `A` type in this function isn't the `A` type defined in the class so if you don't declare `<A: Any>` in the function the compiler will tell you "no no no"